examples/basic_motion.py

```diff
@@ -11,43 +11,69 @@
 
 import numpy as np
 
-from opensourceleg.osl import OpenSourceLeg
-from opensourceleg.tools import units
+from opensourceleg.actuators.dephy import DephyActuator
+from opensourceleg.time import SoftRealtimeLoop
+from opensourceleg.units import units
 
-osl = OpenSourceLeg(frequency=200)
-osl.add_joint("knee", gear_ratio=9 * 83 / 18)
-osl.add_joint("ankle", gear_ratio=9 * 83 / 18)
+FREQUENCY = 200
 
+knee = DephyActuator(
+    tag="knee",
+    firmware_version="7.2.0",
+    port="/dev/ttyACM0",
+    gear_ratio=9 * 83 / 18,
+    frequency=FREQUENCY,
+)
 
-def make_periodic_traj_func(period, minimum, maximum):
+ankle = DephyActuator(
+    tag="ankle",
+    firmware_version="7.2.0",
+    port="/dev/ttyACM1",
+    gear_ratio=9 * 83 / 18,
+    frequency=FREQUENCY,
+)
+
+acutators = [knee, ankle]
+
+clock = SoftRealtimeLoop(dt=1 / FREQUENCY)
+
+
+def make_periodic_trajectory(period, minimum, maximum):
     amplitude = (maximum - minimum) / 2
     mean = amplitude + minimum
     return lambda t: amplitude * np.cos(t * 2 * np.pi / period) + mean
 
 
-ankle_traj = make_periodic_traj_func(10, -20, 20)
-knee_traj = make_periodic_traj_func(10, 10, 90)
+ankle_traj = make_periodic_trajectory(10, -20, 20)
+knee_traj = make_periodic_trajectory(10, 10, 90)
+
+with knee, ankle:
+    for actuator in acutators:
+        actuator.home()
 
-with osl:
-    osl.home()
     input("Homing complete: Press enter to continue")
-    osl.knee.set_mode(osl.knee.control_modes.position)
-    osl.ankle.set_mode(osl.ankle.control_modes.position)
-    osl.knee.set_position_gains(kp=5)
-    osl.ankle.set_position_gains(kp=5)
 
-    for t in osl.clock:
-        osl.update()
+    knee.set_control_mode(knee.CONTROL_MODES.POSITION)
```

Comment:
Do we have a check to raise errors if someone tries to do 
``` 
knee.set_control_mode(ankle.CONTROL_MODES.POSITION)```?

Reply:
You're gonna appreciate this: There are no actuator (instance) specific CONTROL_MODES now!!

Previously each control mode class had a reference to the actuator with which it used to set gains and motor commands.

I've refactored this setup (which is potentially problematic as it takes away power from the actuator class and gives it to each control mode class) to use a global CONTROL_MODES enum that only holds an integer flag to differentiate / identify modes. The actuator class is now the object that interacts closely with the hardware and not the control mode classes, which was the root cause of the issue that you mentioned.

Reference: [code](https://github.com/neurobionics/opensourceleg/blob/8caf8ce509ba5d945f3116492b69a3cbdfc65442/opensourceleg/actuators/dephy.py)

Reply:
Sweet, this makes more sense to me anyway. Nice work!